src/stan/services/util/initialize.hpp

```diff
@@ -13,202 +13,211 @@
-            if (msg.str().length() > 0)
-              logger.info(msg);
-          } catch (std::domain_error& e) {
-            if (msg.str().length() > 0)
-              logger.info(msg);
-            logger.info("Rejecting initial value:");
-            logger.info("  Error evaluating the log probability"
-                        " at the initial value.");
-            logger.info(e.what());
-            continue;
-          } catch (std::exception& e) {
-            if (msg.str().length() > 0)
-              logger.info(msg);
-            logger.info("Unrecoverable error evaluating the log probability"
-                        " at the initial value.");
-            logger.info(e.what());
-            throw;
-          }
-          if (!boost::math::isfinite(log_prob)) {
-            logger.info("Rejecting initial value:");
-            logger.info("  Log probability evaluates to log(0),"
-                        " i.e. negative infinity.");
-            logger.info("  Stan can't start sampling from this"
-                        " initial value.");
-            continue;
-          }
-          std::stringstream log_prob_msg;
-          std::vector<double> gradient;
-          bool gradient_ok = true;
-          clock_t start_check = clock();
-          try {
-            log_prob = stan::model::log_prob_grad<true, true>
-              (model, unconstrained, disc_vector,
-               gradient, &log_prob_msg);
-          } catch (const std::exception& e) {
-            if (log_prob_msg.str().length() > 0)
-              logger.info(log_prob_msg);
-            logger.info(e.what());
-            throw;
-          }
-          clock_t end_check = clock();
-          double deltaT = static_cast<double>(end_check - start_check)
-            / CLOCKS_PER_SEC;
-          if (log_prob_msg.str().length() > 0)
-            logger.info(log_prob_msg);
+      if (!any_initialized) {
+        unconstrained = random_context.get_unconstrained();
+      } else {
+        stan::io::chained_var_context context(init, random_context);
 
-          for (size_t i = 0; i < gradient.size(); ++i) {
-            if (gradient_ok && !boost::math::isfinite(gradient[i])) {
-              logger.info("Rejecting initial value:");
-              logger.info("  Gradient evaluated at the initial value"
-                          " is not finite.");
-              logger.info("  Stan can't start sampling from this"
-                          " initial value.");
-              gradient_ok = false;
-            }
-          }
-          if (gradient_ok && print_timing) {
-            logger.info("");
-            std::stringstream msg1;
-            msg1 << "Gradient evaluation took " << deltaT << " seconds";
-            logger.info(msg1);
+        model.transform_inits(context,
+                              disc_vector,
+                              unconstrained,
+                              &msg);
+      }
+    } catch (std::domain_error& e) {
```

Comment:
Did someone take a pass to make sure we only throw domain errors when we want to reject and keeping going?  I thought we still had Boost errors of various types and other kinds of `std::exception` like range errors and things that shouldn't stop sampling.

Reply:
I did a while ago. After this catch, we catch all `std::exception`, which include Boost errors.

I haven't seen any behavior stop sampling where it shouldn't. Have you? Or have you seen it reported anywhere? We can take a pass over every throw if you want.